manta-pay/abstract-protocol.tex

```diff
@@ -192,146 +288,230 @@ \subsection{Abstract Cryptographic Schemes}
 
 We begin by defining the cryptographic primitives involved in the \Transfer{} protocol:
 
 \begin{definition}[Transfer Configuration]
     A \TransferConfiguration{} is a collection of implementations of the following abstract cryptographic primitives:
     \begin{itemize}
-        \item \textbf{Hybrid Public Key Encryption}: $\HPKE$
+        \item \textbf{Key Schedule}: $\KeySchedule$
+        \item \textbf{Incoming Authenticated Hybrid Public Key Encryption}: $\aHPKEin$
+        \item \textbf{Outgoing Authenticated Hybrid Public Key Encryption}: $\aHPKEout$
         \item \textbf{UTXO Commitment Scheme}: $\COM^{\UTXO}$
+        \item \textbf{HASH Function}: $\HASH{}$
         \item \textbf{Void Number Commitment Scheme}: $\COM^{\VN}$
         \item \textbf{Dynamic Cryptographic Accumulator}: $\DCA$
         \item \textbf{Zero-Knowledge Proving System}: $\NIZK$
     \end{itemize}
     with the following notational conventions:
     \begin{align*}
-        \KA            &:= \HPKE.\KA \\
-        \UTXO          &:= \COM^{\UTXO}.\Output \\
-        \VoidNumber    &:= \COM^{\VN}.\Output \\
-        \EncryptedNote &:= \HPKE.\Message \\
-        \UTXOSet       &:= \DCA
+        \UTXO         &:= \COM^{\UTXO}.\Output \\
+        \TUTXO        &:= \PublicAsset \times \UTXO \times \Bool \times \tUTXOID \\
+        \VoidNumber   &:= \COM^{\VN}.\Output \\
+        \IncomingNote &:= \KeySchedule.\ShieldedAddress \times \COM^\UTXO.\Randomness \times \Asset \\
+        \OutgoingNote &:= \Asset \\
+        \TUTXOSet     &:= \DCA
     \end{align*}
     and the following constraints:
     \begin{align*}
-        \COM^{\UTXO}.\Randomness &= \KA.\SecretKey \\
-        \COM^{\UTXO}.\Input      &= \KA.\PublicKey \times \Asset \\
-        \COM^{\VN}.\Randomness   &= \KA.\SecretKey \\
-        \COM^{\VN}.\Input        &= \UTXO \\
-        \UTXOSet.\Item           &= \UTXO \\
-        \ValidTransfer           &: \NIZK.\Statement
+        \COM^{\UTXO}.\Input    &= \KeySchedule.\ShieldedAddress \times \Asset \\
+        \COM^{\VN}.\Randomness &= \KeySchedule.\ProofAuthorizingKey \\
+        \COM^{\VN}.\Input      &= \UTXO \\
+        \TUTXOSet.\Item        &= \TUTXO \\
+        \aHPKEin.\KA           &= \KeySchedule.\KA \\
+        \aHPKEout.\KA          &= \KeySchedule.\KA \\
+        \ValidTransfer         &: \NIZK.\Statement \\
```

Comment:
If I understood this protocol correctly, we might need the additional constraint that tUTXOSet.insert = HASH.hash (with the implied relations on the domains and targets of these maps).

Reply:
@SupremoUGH